### PR TITLE
Implement async storage support

### PR DESCRIPTION
**Pull request by RichiCoder1**
Adds promise support to [save|restore]State and asynchronous storage provider support.

Adds explicit tests for localforage support.

Apologies for some of the whitespace changes. Couldn't figure out how to turn off some of them in vscode.

Closes #10 ;

### CODE REVIEW

File: src/SimplePromiseQueue.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+export default class SimplePromiseQueue {
+    private readonly _queue: Promise<void>[] = [];
```
Comment (championswimmer) [inline src/SimplePromiseQueue.ts]: Can you use Array<Promise<void>> instead ? (stickler for Typescript non-simple-types) :P

Comment (championswimmer) [reply src/SimplePromiseQueue.ts]: And no semicolon please

File: src/index.ts [hunk 2]
```diff
@@ -1,32 +1,39 @@
 /**
  * Created by championswimmer on 18/07/17.
  */
-import {Mutation, Payload, Plugin, Store} from 'vuex'
+import { Mutation, Payload, Plugin, Store } from 'vuex'
 import MockStorage from './MockStorage'
+import SimplePromiseQueue from './SimplePromiseQueue'
 import merge from 'lodash.merge'
+
+export type AsyncStorage = {
```
Comment (championswimmer) [inline src/index.ts]: Use interface please

**Comment by RichiCoder1** [comment]
Apologies I didn't get to those fixes sooner!

**Pull request merged**